### DISCUSSION TITLE
[3차원 그래픽스]챕터2 연습문제 1번 풀이

### DISCUSSION THREAD

**Discussion by Lcat12345** [MyeongWoonJang/cpp-study#101] [Show and tell]
문제
---
> 1. 2차원에서 (원점 대신) 특정 점 ( a, b )를 중심으로 θ 만큼 회전하는 아핀 변환 행렬을 계산하라.

풀이
---
원점으로 이동해서 회전한다
( a, b )를 중심으로 회전하므로 이동행렬은 아래와 같다.

![image](https://user-images.githubusercontent.com/62102830/201528518-a4d20834-a329-4cb1-ad8f-bf922e93c3a4.png)

회전행렬을 동차좌표계로 변환 후 최종 아핀 변환 행렬을 구하면된다.

![image](https://user-images.githubusercontent.com/62102830/201528552-5aa494db-9579-4d9e-b3e3-665002e87f21.png)

**Comment by Woon-2** [comment]
컴퓨터 그래픽스에서 비슷한 내용을 다룬 적이 있었죠.   
임의의 회전축에 대한 회전은 어떻게 표현하냐! 하고요.

(a, b)가 중심이니 원점을 (a, b)로 옮긴 뒤 회전시키신 건 훌륭했습니다.   

>"원점을 옮겼다." 라고 표현한 것은,    
>f: X → Y일 때 x ∈ X, y ∈ Y, f(x, y) = 0을 만족하는 임의의 x, y에 대해 모든 순서쌍 (x, y)가 g: (x, y)↦(x-a, y-b)를 거쳐   
>f(x+a, y+b) = 0를 만족시키도록 했기 때문입니다.   
>
>가볍게 말하면 평행이동이죠.   
>좌표는 원점에 대한 상대적인 위치이니, 모든 좌표를 (-a, -b)만큼 이동시킨 것은   
>원점을 (a, b)만큼 이동시킨 것과 같습니다.

그런데, 제자리에 돌려놓질 않았네요.   
회전시키고 난 뒤엔 다시 원점을 (0, 0)으로 옮겨야 합니다.   
무엇이 필요한지 아시겠죠?

**Reply by Lcat12345** [reply]
다시 (a, b)만큼 이동하는 이동행렬을 곱해보았습니다

![image](https://user-images.githubusercontent.com/62102830/201651038-79c2fde8-d684-44d5-bb9d-321ea9688422.png)
